Frameworks/UIKit/UITextField.mm

```diff
@@ -129,13 +129,17 @@ - (void)setText:(NSString*)text {
 
         [_secureModeLock lock];
         if (_secureTextMode) {
-            _passwordBox.password = _text;
+            if (![_passwordBox.password isEqualToString:_text]) {
+                _passwordBox.password = _text;
+            }
         } else {
-            _textBox.text = _text;
-            // Ensure caret at end of field in case we programmatically
-            // gain focus (becomeFirstResponder) after the text is set:
-            _textBox.selectionStart = [_text length];
-            _textBox.selectionLength = 0;
+            if (![_textBox.text isEqualToString:_text]) {
```

Comment:
> if (![_textBox.text isEqualToString:_text]) { [](start = 11, length = 46)

It seems like the problem is due to us calling setText in the textchanged handler.  If we just apply the Xaml controls' templates, can we get rid of all of these cached values?  That would set us up for an easier transition to a Xaml control implementation, too.